src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -198,8 +197,15 @@ public TimeSpan NonceLifetime
         /// <summary>
         /// Gets or sets a value indicating if a 'sub' claim is required.
         /// </summary>
-        [DefaultValue(false)]
-        public bool RequireSub { get; set; }
+        [DefaultValue(true)]
+        public bool RequireSub { get; set; } = RequireSubByDefault;
```

Comment:
Is changing this from optional to required going to break compatibility with other IDPs that might not provide the subject in some cases? Can we confidently do this without breaking anyone's existing code?